services/frontend-service/src/ui/ActionsCart/CheckoutDialog.tsx

```diff
@@ -78,11 +75,75 @@ const ApplyButton = (props: { openNotification: (msg: string) => void; closeDial
     }
 };
 
-export const CheckoutCart = () => {
-    const [openNotify, setOpenNotify] = React.useState(false);
-    const [dialogOpen, setDialogOpen] = React.useState(false);
-    const [notifyMessage, setNotifyMessage] = React.useState('');
+const showTextField = (actions: BatchAction[]) => {
+    for (const act of actions) {
+        const t = GetActionDetails(act).type;
+        if (t === ActionTypes.CreateEnvironmentLock || t === ActionTypes.CreateApplicationLock) {
+            return true;
+        }
+    }
+    return false;
+};
```

Comment:
Thank you for suggesting these functions and for including the links, I hadn't used them before! 

It works, but I still had to extract the type from `lockActions` because the `.includes` function also complained :(